addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1278,17 +1279,8 @@ const Wysiwyg = Widget.extend({
                         anchorOffset = focusOffset = index;
                     }
                 } else {
-                    const isDirectionRight = getCursorDirection(selection.anchorNode, 0, selection.focusNode, 0) === DIRECTIONS.RIGHT;
-                    if (
-                        closestElement(selection.anchorNode, 'a') === link &&
-                        closestElement(selection.focusNode, 'a') === link
-                    ) {
-                        [anchorNode, focusNode] = isDirectionRight
-                            ? [selection.anchorNode, selection.focusNode]
-                            : [selection.focusNode, selection.anchorNode];
-                    } else {
-                        [anchorNode, focusNode] = [link, link];
-                    }
+                    const commonBlock = selection.rangeCount && closestBlock(selection.getRangeAt(0).commonAncestorContainer);
+                    [anchorNode, focusNode] = link.contains(commonBlock) ? [commonBlock, commonBlock] : [link, link];
```

Comment:
The code that you removed was part of a bugfix. Did you blame it and try if you could not reproduce the original issue specified in the commit message? https://github.com/odoo/odoo/pull/145925/commits/cf9148c4ce4937cfa65ae6467fabe6d82b931076

Reply:
Yes. I linked to the commit in the commit message. I modified his fix so that instead of selecting the anchor and focus, I select the closest common block if it's in the link, and the link otherwise. It's just limiting his fix to the case he tried to fix.